index.js

```diff
@@ -1039,6 +1039,66 @@
       [Fn(a, $.Boolean), m(a), m(a)],
       Z.filterM);
 
+  //# takeWhile :: (Foldable f, Monoid (f a)) => (a -> Boolean) -> f a -> f a
```

Comment:
I think we should replace `Monoid (f a)` with `Alternative f` for the reason given in sanctuary-js/sanctuary-type-classes#37.

Reply:
I think we updated the type signature without switching from `Z.empty` to `Z.zero`. I'll rectify this in the pull request I'll soon open for sanctuary-type-classes.

Reply:
Actually, given our `concat`-based implementation I think `Monoid (f a)` is the correct constraint after all. We could have a `chain`-based `takeWhileM` (to match `filterM`) but it seems excessive to provide two versions of a function which is only used occasionally.

Reply:
sanctuary-js/sanctuary-type-classes#69